### PR TITLE
dcir example in hfss3dlayout

### PR DESCRIPTION
**Pull request by ring630**

### CODE REVIEW

File: examples/02-HFSS3DLayout/Dcir_in_3DLayout.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    name="U2A5-V3P3_S5"
+)
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# add SIwave DCIR analysis
+
+appedb.core_siwave.add_siwave_dc_analysis(name="my_setup")
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Save and close Edb
```
Comment (MaxJPRey) [inline examples/02-HFSS3DLayout/Dcir_in_3DLayout.py:77]:
```suggestion
################
# Save and close EDB
# ~~~~~~~~~~~~~
# Save and close EDB.
```

File: examples/02-HFSS3DLayout/Dcir_in_3DLayout.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+appedb.save_edb()
+appedb.close_edb()
+
+###############################################################################
+# Analysis DCIR in AEDT
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Launch AEDT and import the configured EDB and analysis DCIR
```
Comment (MaxJPRey) [inline examples/02-HFSS3DLayout/Dcir_in_3DLayout.py:85]:
```suggestion
##################
# Analysis DCIR in AEDT
# ~~~~~~~~~~~~~~
# Launch AEDT and import the configured EDB and analysis DCIR
```

File: examples/02-HFSS3DLayout/Dcir_in_3DLayout.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Get via current from dcir result
+via_current = hfss3dl.get_dcir_solution_data(
+    setup_name="my_setup",
+    show="Vias",
+    category="Current")
+print({ expression: via_current.data_magnitude(expression) for  expression in via_current.expressions[100:105]})
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Close AEDT
```
Comment (MaxJPRey) [inline examples/02-HFSS3DLayout/Dcir_in_3DLayout.py:116]:
```suggestion
###############################################################################
# Close AEDT
# ~~~~~~~
```

File: examples/02-HFSS3DLayout/Dcir_in_3DLayout.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Get voltage from dcir result
+voltage = hfss3dl.get_dcir_solution_data(
+    setup_name="my_setup",
+    show="Sources",
+    category="Voltage")
+print({ expression: voltage.data_magnitude(expression) for  expression in voltage.expressions})
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (MaxJPRey) [inline examples/02-HFSS3DLayout/Dcir_in_3DLayout.py:107]:
```suggestion
###############################################################################
# Get via
# ~~~~~
```

File: examples/02-HFSS3DLayout/Dcir_in_3DLayout.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+# Get loop resistance from dcir result
+loop_resistance = hfss3dl.get_dcir_solution_data(
+    setup_name="my_setup",
+    show="RL",
+    category="Loop_Resistance")
+print({expression: loop_resistance.data_magnitude(expression) for expression in loop_resistance.expressions})
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (MaxJPRey) [inline examples/02-HFSS3DLayout/Dcir_in_3DLayout.py:99]:
```suggestion
###############################################################################
# Get voltage
# ~~~~~~~~
```

File: examples/02-HFSS3DLayout/Dcir_in_3DLayout.py [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+###############################################################################
+# Analysis DCIR in AEDT
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Launch AEDT and import the configured EDB and analysis DCIR
+desktop = pyaedt.Desktop(edbversion, non_graphical=False, new_desktop_session=True)
+hfss3dl = pyaedt.Hfss3dLayout(local_path)
+hfss3dl.analyze_all()
+hfss3dl.save_project()
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (MaxJPRey) [inline examples/02-HFSS3DLayout/Dcir_in_3DLayout.py:91]:
```suggestion
###############################################################################
# Get loop resistance
# ~~~~~~~~~~~~~
```

File: examples/02-HFSS3DLayout/Dcir_in_3DLayout.py [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+# Configure Edb for DCIR analysis
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Copy example into temporary folder
+temp_dir = tempfile.gettempdir()
+dst_dir = os.path.join(temp_dir, pyaedt.generate_unique_name("pyaedt_dcir"))
+os.mkdir(dst_dir)
+local_path = pyaedt.downloads.download_aedb(dst_dir)
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Load example board into pyaedt.Edb
```
Comment (PipKat) [inline examples/02-HFSS3DLayout/Dcir_in_3DLayout.py:15]:
```suggestion
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# Load example board into EDB
```

File: examples/02-HFSS3DLayout/Dcir_in_3DLayout.py [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Create pin group on VRM negative pins
+
+appedb.core_siwave.create_pin_group_on_net(
+    reference_designator="U3A1",
+    net_name="GND",
+    group_name="U3A1-GND")
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Create place voltage source between VRM positive and negative pin groups
```
Comment (PipKat) [inline examples/02-HFSS3DLayout/Dcir_in_3DLayout.py:38]:
```suggestion
# Create voltage source between VRM positive and negative pin groups
```

File: examples/02-HFSS3DLayout/Dcir_in_3DLayout.py [hunk 9]
```diff
@@ -0,0 +1,127 @@
+import os
```
Comment (maxcapodi78) [inline examples/02-HFSS3DLayout/Dcir_in_3DLayout.py:1]: We need the header in the example to be able to process with Sphinx. you can take it from one of other examples.

File: examples/02-HFSS3DLayout/Dcir_in_3DLayout.py [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Create pin group on sink component negative pins
+
+appedb.core_siwave.create_pin_group_on_net(
+    reference_designator="U2A5",
+    net_name="GND",
+    group_name="U2A5-GND")
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (maxcapodi78) [inline examples/02-HFSS3DLayout/Dcir_in_3DLayout.py:62]:
```suggestion
#####################################################################################
```

File: examples/02-HFSS3DLayout/Dcir_in_3DLayout.py [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Create pin group on sink component positive pins
+
+appedb.core_siwave.create_pin_group_on_net(
+    reference_designator="U2A5",
+    net_name="V3P3_S5",
+    group_name="U2A5-V3P3_S5")
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (maxcapodi78) [inline examples/02-HFSS3DLayout/Dcir_in_3DLayout.py:54]:
```suggestion
#####################################################################################
```

File: examples/02-HFSS3DLayout/Dcir_in_3DLayout.py [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Create voltage source between VRM positive and negative pin groups
+appedb.core_siwave.create_voltage_source_on_pin_group(
+    pos_pin_group_name="U3A1-BST_V3P3_S5",
+    neg_pin_group_name="U3A1-GND",
+    magnitude=3.3,
+    name="U3A1-BST_V3P3_S5"
+)
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (maxcapodi78) [inline examples/02-HFSS3DLayout/Dcir_in_3DLayout.py:46]:
```suggestion
#####################################################################################
```

File: examples/02-HFSS3DLayout/Dcir_in_3DLayout.py [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+# Create pin group on VRM negative pins
+
+appedb.core_siwave.create_pin_group_on_net(
+    reference_designator="U3A1",
+    net_name="GND",
+    group_name="U3A1-GND")
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (maxcapodi78) [inline examples/02-HFSS3DLayout/Dcir_in_3DLayout.py:37]:
```suggestion
#####################################################################################
```

File: examples/02-HFSS3DLayout/Dcir_in_3DLayout.py [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Create pin group on VRM positive pins
+
+gnd_name = "GND"
+appedb.core_siwave.create_pin_group_on_net(
+    reference_designator="U3A1",
+    net_name="BST_V3P3_S5",
+    group_name="U3A1-BST_V3P3_S5")
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (maxcapodi78) [inline examples/02-HFSS3DLayout/Dcir_in_3DLayout.py:29]:
```suggestion
#####################################################################################
```

File: examples/02-HFSS3DLayout/Dcir_in_3DLayout.py [hunk 15]
```diff
@@ -0,0 +1,127 @@
...
+os.mkdir(dst_dir)
+local_path = pyaedt.downloads.download_aedb(dst_dir)
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Load example board into EDB
+
+edbversion = "2022.2"
+appedb = pyaedt.Edb(local_path, edbversion=edbversion)
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (maxcapodi78) [inline examples/02-HFSS3DLayout/Dcir_in_3DLayout.py:20]:
```suggestion
#####################################################################################
```

File: examples/02-HFSS3DLayout/Dcir_in_3DLayout.py [hunk 16]
```diff
@@ -0,0 +1,127 @@
...
+###############################################################################
+# Configure EDB for DCIR analysis
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Copy example into temporary folder
+temp_dir = tempfile.gettempdir()
+dst_dir = os.path.join(temp_dir, pyaedt.generate_unique_name("pyaedt_dcir"))
+os.mkdir(dst_dir)
+local_path = pyaedt.downloads.download_aedb(dst_dir)
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (maxcapodi78) [inline examples/02-HFSS3DLayout/Dcir_in_3DLayout.py:14]:
```suggestion
#####################################################################################
```

File: examples/02-HFSS3DLayout/Dcir_in_3DLayout.py [hunk 17]
```diff
@@ -0,0 +1,127 @@
...
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Create place current source between sink component positive and negative pin groups
+appedb.core_siwave.create_current_source_on_pin_group(
+    pos_pin_group_name="U2A5-V3P3_S5",
+    neg_pin_group_name="U2A5-GND",
+    magnitude=1,
+    name="U2A5-V3P3_S5"
+)
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (maxcapodi78) [inline examples/02-HFSS3DLayout/Dcir_in_3DLayout.py:71]:
```suggestion
###############################################################################
```

**Review by maxcapodi78 — APPROVED** [review]

**Pull request merged**